pkg/test/proxy/envoy/envoy.go

```diff
@@ -0,0 +1,212 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package envoy
+
+import (
+	"fmt"
+	"go/build"
+	"math/rand"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+)
+
+const (
+	// DefaultLogLevel the log level used for Envoy if not specified.
+	DefaultLogLevel = LogLevelTrace
+	maxBaseID       = 1024 * 16
+)
+
+// LogLevel represents the log level to use for Envoy.
+type LogLevel string
+
+const (
+	// LogLevelTrace level
+	LogLevelTrace LogLevel = "trace"
+	// LogLevelDebug level
+	LogLevelDebug LogLevel = "debug"
+	// LogLevelInfo level
+	LogLevelInfo LogLevel = "info"
+	// LogLevelWarning level
+	LogLevelWarning LogLevel = "warning"
+	// LogLevelError level
+	LogLevelError LogLevel = "error"
+	// LogLevelCritical level
+	LogLevelCritical LogLevel = "critical"
+	// LogLevelOff level
+	LogLevelOff = "off"
+)
+
+// Envoy is a wrapper that simplifies running Envoy.
+type Envoy struct {
+	// ConfigFile (required) the v2 config file for Envoy.
+	ConfigFile string
+	// BinPath (optional) the path to the Envoy binary. If not set, uses the debug binary under ISTIO_OUT. If the
+	// ISTIO_OUT environment variable is not set, the default location under GOPATH is assumed. If ISTIO_OUT contains
+	// multiple debug binaries, the most recent file is used.
+	BinPath string
+	// LogFilePath (optional) Sets the output log file for Envoy. If not set, Envoy will output to stderr.
+	LogFilePath string
+	// LogLevel (optional) if provided, sets the log level for Envoy. If not set, DefaultLogLevel will be used.
+	LogLevel LogLevel
+	cmd      *exec.Cmd
+}
+
+// Start starts the Envoy process.
+func (e *Envoy) Start() error {
```

Comment:
Can we use the 'Start' in pkg/bootstrap ? Or the one in tests/mixer ?

Reply:
What do you mean exactly? Are you referring to the style of the call, where instead of start/stop we have start(chan)?